### ISSUE TITLE
Fail to remove IoT Resources

### ISSUE THREAD

**Issue by EliassonHenrik** [rebuy-de/aws-nuke#374]
I've noticed that the following targets have issues, preventing IoT-resources to be deleted:
- IoTCertificate
- IoTPolicy
- IoTThing (x2)

In other words, all of these targets needs to implement some preparations before the actual delete can be performed.

**IoTCertificate**
Certificates can not be deleted if they are `ACTIVE`, they need to be set to `INACTIVE` before.
If not, `DeleteCertificate` action get this error:
```
CertificateStateException
Certificate must be deactivated (not ACTIVE) before deletion. Id: <certificate>
```

**IoTPolicy**
Policies can not be deleted if they have certificates attached, they need to be detached before.
If not, `DeletePolicy` action get this error:
```
DeleteConflictException
The policy cannot be deleted as the policy is attached to one or more principals (name=<policy>)
```

**IoTThing**
Things can not be deleted if they have certificates attached, they need to be detached before. 
If not, `DeleteThing` action get this error:
```
InvalidRequestException
Cannot delete. Thing <thing> is still attached to one or more principals
```

**IoTThing**
Things can not be deleted if they have versions available, versions need to be deleted before. 
If not, `DeletePolicy` action get this error:
```
DeleteConflictException
The policy cannot be deleted as versions other than than default version exist (name=<policy>)
```

**Comment by bjoernhaeuser** [comment]
@EliassonHenrik would you be interested in providing the necessary changes?

**Comment by npellegrin** [comment]
Was the issue fixed by #392 ? If yes, this issue could be closed.

**Comment by der-eismann** [comment]
Fixed by #392
